File: packages/aws-cdk/test/commands/test-resources/stacks/s3_stack.py
from aws_cdk import Stack
import aws_cdk as cdk
import aws_cdk.aws_s3 as s3
from constructs import Construct

"""
  AWS CloudFormation Sample Template S3_Website_Bucket_With_Retain_On_Delete: Sample template showing how to create a publicly accessible S3 bucket configured for website access with a deletion policy of retain on delete.
"""
class GoodPythonStack(Stack):
  def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
    super().__init__(scope, construct_id, **kwargs)

    # Resources
    s3Bucket = s3.CfnBucket(self, 'S3Bucket',
          access_control = 'PublicRead',
          website_configuration = {
            'indexDocument': 'index.html',
            'errorDocument': 'error.html',
          },
        )
    s3Bucket.cfn_options.deletion_policy = cdk.CfnDeletionPolicy.RETAIN

    # Outputs
    """
      URL for website hosted on S3
    """
    self.website_url = s3Bucket.attr_website_url
    cdk.CfnOutput(self, 'CfnOutputWebsiteURL', 
      key = 'WebsiteURL',
      description = 'URL for website hosted on S3',
      value = str(self.website_url),
    )

    """
      Name of S3 bucket to hold website content
    """
    self.s3_bucket_secure_url = ''.join([
      'https://',
      s3Bucket.attr_domain_name,
    ])
    cdk.CfnOutput(self, 'CfnOutputS3BucketSecureURL', 
      key = 'S3BucketSecureURL',
      description = 'Name of S3 bucket to hold website content',
      value = str(self.s3_bucket_secure_url),
    )

File: packages/aws-cdk/test/commands/test-resources/stacks/s3-stack.ts
import * as cdk from 'aws-cdk-lib';
import * as s3 from 'aws-cdk-lib/aws-s3';

export interface GoodTypeScriptStackProps extends cdk.StackProps {
}

/**
 * AWS CloudFormation Sample Template S3_Website_Bucket_With_Retain_On_Delete: Sample template showing how to create a publicly accessible S3 bucket configured for website access with a deletion policy of retain on delete.
 */
export class GoodTypeScriptStack extends cdk.Stack {
  /**
   * URL for website hosted on S3
   */
  public readonly websiteUrl;
  /**
   * Name of S3 bucket to hold website content
   */
  public readonly s3BucketSecureUrl;

  public constructor(scope: cdk.App, id: string, props: GoodTypeScriptStackProps = {}) {
    super(scope, id, props);

    // Resources
    const s3Bucket = new s3.CfnBucket(this, 'S3Bucket', {
      accessControl: 'PublicRead',
      websiteConfiguration: {
        indexDocument: 'index.html',
        errorDocument: 'error.html',
      },
    });
    s3Bucket.cfnOptions.deletionPolicy = cdk.CfnDeletionPolicy.RETAIN;

    // Outputs
    this.websiteUrl = s3Bucket.attrWebsiteUrl;
    new cdk.CfnOutput(this, 'WebsiteURL', {
      description: 'URL for website hosted on S3',
      value: this.websiteUrl!.toString(),
    });
    this.s3BucketSecureUrl = [
      'https://',
      s3Bucket.attrDomainName,
    ].join('');
    new cdk.CfnOutput(this, 'S3BucketSecureURL', {
      description: 'Name of S3 bucket to hold website content',
      value: this.s3BucketSecureUrl!.toString(),
    });
  }
}
